### PR TITLE
Issue 76 boundary conditions bug

### LINKED ISSUE #76: Bug when using model.rhs.evaluate()
There is an issue using the `model.rhs.evaluate()` function which depends on the form of the flux `N_e` in the electrolyte model which is indicated by an arrow in the code below: 

```
       epsilon = pybamm.standard_parameters.epsilon_s  
        b = pybamm.standard_parameters.b
        delta = pybamm.standard_parameters.delta
        nu = pybamm.standard_parameters.nu
        t_plus = pybamm.standard_parameters.t_plus
        ce0 = pybamm.standard_parameters.ce0

        electrolyte_domain = ["whole cell"]
        c_e = pybamm.Variable("c_e", electrolyte_domain)

        N_e = -pybamm.grad(c_e)  <--------------------------------------------------(1)
        self.rhs = {
            c_e: -pybamm.div(N_e) / delta / epsilon + nu * (1 - t_plus) * G
        }  # / delta / epsilon + nu * (1 - t_plus) * G
        self.initial_conditions = {c_e: ce0}
        self.boundary_conditions = {
            N_e: {"left": pybamm.Scalar(0), "right": pybamm.Scalar(0)}
        } 
```
Within `test_electrolyte.py` I have
```
        G = pybamm.Scalar(1)
        model = pybamm.electrolyte.StefanMaxwellDiffusion(G)

        param = pybamm.ParameterValues(
            "input/parameters/lithium-ion/parameters/LCO.csv"
        )

        param.process_model(model)

        mesh = pybamm.FiniteVolumeMacroMesh(param, 2)
        disc = pybamm.FiniteVolumeDiscretisation(mesh)
        disc.process_model(model)

        y0 = model.initial_conditions
        model.rhs.evaluate(None, y0) <---------------------------------------(2)
```
However, if i change (1) to `N_e = epsilon * pybamm.grad(c_e)` , there is an error at (2). Link to the failed test output: https://gist.github.com/Scottmar93/b7bcbff685383d9a5dceaca630f3aa40

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Process parameters for boundary conditions so that they are recognised when discretising if a Parameter changes to a Scalar (see #76 for clearer description).

Fixes #76 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_models/test_electrolyte.py [hunk 1]
```diff
@@ -33,8 +34,37 @@ def test_processing_disc(self):
...
+        y0 = model.concatenated_initial_conditions
+        np.testing.assert_array_equal(y0, np.ones_like(mesh["whole cell"].nodes))
+        print("model rhs is: ", model.concatenated_rhs.evaluate(None, y0))
+
+        # Solve
+        # solver = pybamm.ScipySolver(tol=1e-8, method="RK45")
+        # t_eval = mesh["time"]
+        # print("This is the solver: ", solver)
+        # print("This is t_eval: ", t_eval)
+        # solver.solve(model, t_eval)
```
Comment (Scottmar93) [inline tests/test_models/test_electrolyte.py]: run the solver tests
